### DISCUSSION TITLE
Enable OS notifications for vscode commands that use sf/sfdx cli commands

### DISCUSSION THREAD

**Discussion by AllanOricil** [forcedotcom/salesforcedx-vscode#5438] [Ideas]
This feature would need to be implemented after this [one](https://github.com/forcedotcom/cli/discussions/2730)

## Why? 

Vscode notifications are bound to vscode client only, according to their [docs](https://code.visualstudio.com/api/ux-guidelines/notifications).

## Value

Improves UX for developers that work with multiple vscode instances, which might be hidden while executing a **synchronous** sf command. For example:

- developer can start a **synchronous** bulk upload job using a huge file
- developer can start a **synchronous** metadata validate/deploy job

When a synch command finishes, a vscode, or OS, notification would notify the User his job is done.

## How?

I believe that if these vscode commands are just calling sfdx commands in the background, it will be just a matter of setting up an env variable called `SF_NOTIFY` to true in order to ask the cli to notify its caller. This env variable can be set based on a vscode extension setting using [`contributes.configuration`](https://code.visualstudio.com/api/references/contribution-points#contributes.configuration). But of course you would need to have the notification system implemented in the cli.

**Comment by AllanOricil** [comment]
@AnanyaJha

**Reply by AnanyaJha** [reply]
Thanks for the feature request @AllanOricil ! We'll work with the CLI team internally to investigate.

**Comment by AllanOricil** [comment]
relates https://github.com/forcedotcom/cli/discussions/2742
